internal/app/go-filecoin/node/node.go

```diff
@@ -693,6 +664,11 @@ func (node *Node) CreateMiningWorker(ctx context.Context) (*mining.DefaultWorker
 		return nil, err
 	}
 
+	poster := node.BlockMining.PoStGenerator
+	if poster == nil {
+		poster = node.StorageMining.PoStGenerator
+	}
```

Comment:
The PoStGenerator in BlockMining sometimes gets set to fake proofs, but is not always defined in test code. The PoStGenerator in StorageMining is always defined with real proofs. The worker panics with an undefined poster, but we want it to use fake proofs if we're testing with fake proofs.

This solution is suboptimal, but I think a better solution would require re-working the BlockMining and StorageMining submodule.